src/main/java/edu/byu/cs/service/UserService.java

```diff
@@ -51,16 +49,16 @@ public static Collection<RepoUpdate> adminGetRepoHistory(String repoUrl, String
     }
 
     private static void setRepoUrl(String studentNetId, String repoUrl, String adminNetId) throws BadRequestException, InternalServerException, PriorRepoClaimBlockageException {
+        boolean valid;
         try {
-            if (!isValidRepoUrl(repoUrl)) {
-                throw new BadRequestException("Invalid Github Repo Url. Check if the link is valid and points directly to a Github Repo.");
-            }
-        } catch (BadRequestException e) {
-            throw e;
+            valid = RepoUrlValidator.isValidRepoUrl(repoUrl);
         } catch (Exception e) {
             LOGGER.error("Error cloning repo during repoPatch: {}", e.getMessage());
             throw new InternalServerException("There was an internal server error in verifying the Github Repo", e);
         }
+        if (!valid) {
```

Comment:
I only kept the exception block because it was already there for what is essentially the same logic, but I can remove it if we think it wasn't necessary in the first place, since it's accomplishing the same

Reply:
I'm a little confused on if I should: a) remove the Catch block entirely, b) catch specific-typed exceptions instead, c) use the Logger at all with the exception, or d) something entirely I don't understand.
You've clarified _why_ you want it but I'm not fully getting _what_ to do